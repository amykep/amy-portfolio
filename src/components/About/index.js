import React from 'react';
import aboutImage from "../../images/abstract-woman-face.png";

function About()
{
    return (
        <div>

            <section id="about" className="about">
                <h2>About</h2>
                <div className="about-info">
                    <div className="about-img">
                        <img src={aboutImage} alt="Abstract Woman Face" />
                    </div>
                    <div className="about-text">
                        <p>Motivated professional full stack web developer leveraging background in engineering to
                            provide
                            comprehensive solutions. Earned a certificate in Full Stack Web Development from the
                            University
                            of California Berkeley Coding Boot Camp. Innovative problem-solver who is passionate about
                            web
                            design and development. Strengths in creativity, teamwork, and building projects from
                            concept to
                            complete execution.</p>
                    </div>
                </div>
            </section>

        </div>
    );
}

export default About;