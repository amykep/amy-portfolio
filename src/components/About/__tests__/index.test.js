import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import About from '..';


// we'll call the cleanup function using the afterEach global function from Jest. This will ensure that after each test, we won't have any leftover memory data that could give us false results.
afterEach(cleanup);
// Then we'll use the describe function to declare the component we're testing, The first test will be the baseline to verify that the component is rendering. In test two, we'll use the asFragment function, which returns a snapshot of the About component. Then we'll test and compare whether the expected and actual outcomes match. Use the expect function with a matcher to assert something about a value using the toMatchSnapshot matcher to assert that snapshots will match:
describe('About component', () =>
{
    // First Test
    it('renders', () =>
    {
        render(<About />);
    });

    // Second Test
    it('matches snapshot DOM node structure', () =>
    {
        const { asFragment } = render(<About />);
        expect(asFragment()).toMatchSnapshot();
    });
})

