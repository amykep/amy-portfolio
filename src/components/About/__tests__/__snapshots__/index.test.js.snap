// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <div>
    <section
      class="about"
      id="about"
    >
      <h2>
        About
      </h2>
      <div
        class="about-info"
      >
        <div
          class="about-img"
        >
          <img
            alt="Abstract Woman Face"
            src="abstract-woman-face.png"
          />
        </div>
        <div
          class="about-text"
        >
          <p>
            Motivated professional full stack web developer leveraging background in engineering to provide comprehensive solutions. Earned a certificate in Full Stack Web Development from the University of California Berkeley Coding Boot Camp. Innovative problem-solver who is passionate about web design and development. Strengths in creativity, teamwork, and building projects from concept to complete execution.
          </p>
        </div>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
